src/Microsoft.AspNet.Mvc.ViewFeatures/ModelStateDictionaryExtensions.cs

```diff
@@ -0,0 +1,76 @@
+    {
+        /// <summary>
+        /// Adds the specified <paramref name="errorMessage"/> to the <see cref="ModelState.Errors"/> instance
+        /// that is associated with the specified <paramref name="expression"/> text key repressentation.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">The expression of <typeparamref name="TModel"/> from
+        /// which text representation will be used as a key of the <see cref="ModelState"/> to add errors to.</param>
+        /// <param name="errorMessage"></param>
+        public static void AddModelError<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression, string errorMessage)
+        {
+            modelState.AddModelError(GetExpressionText(expression), errorMessage);
+        }
+
+        /// <summary>
+        /// Adds the specified <paramref name="exception"/> to the <see cref="ModelState.Errors"/> instance
+        /// that is associated with the specified <paramref name="expression"/> text key repressentation.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">The expression of <typeparamref name="TModel"/> from
+        /// which text representation will be used as a key of the <see cref="ModelState"/> to add errors to.</param>
+        /// <param name="exception">The error message to add.</param>
+        public static void AddModelError<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression, Exception exception)
+        {
+            modelState.AddModelError(GetExpressionText(expression), exception);
+        }
+
+        /// <summary>
+        /// Removes the specified <paramref name="expression"/> text key repressentation from the <see cref="ModelStateDictionary"/>.
+        /// </summary>
+        /// <typeparam name="TModel">The type of the model.</typeparam>
+        /// <param name="modelState">The <see cref="ModelStateDictionary"/> instance this method extends.</param>
+        /// <param name="expression">The expression of <typeparamref name="TModel"/> from
+        /// which text representation will be used as a key of the <see cref="ModelState"/> to add errors to.</param>
+        /// <returns>
+        /// true if the element is successfully removed; otherwise, false.
+        /// This method also returns false if expression's text key was not found in the model-state dictionary.
+        /// </returns>
+        public static bool Remove<TModel>(this ModelStateDictionary modelState, Expression<Func<TModel, object>> expression)
+        {
+            return modelState.Remove(GetExpressionText(expression));
```

Comment:
Say the current model type is
``` c#
public class MyModel
{
   int A { get; set; }
   int B { get; set; }
   int C { get; set; }
   int D { get; set; }
}
```
Pass `model => model` to this method and it will remove only the `ModelStateDictionary` entry with `Key == string.Empty`. It ignores entries with names "A", "B", "C" and "D".

I'm suggesting users need an option to clean up everything for the given expression. `RemoveAll()` would provide this.